src/sentry/uptime/migrations/0041_uptime_backfill_detector_grouphash.py

```diff
@@ -0,0 +1,102 @@
+# Generated by Django 5.1.7 on 2025-05-06 22:16
+from hashlib import md5
+from typing import Any
+
+from django.db import migrations
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+DATA_SOURCE_UPTIME_SUBSCRIPTION = "uptime_subscription"
+
+
+def get_project_subscription(ProjectUptimeSubscription: Any, detector: Any) -> Any:
+    data_source = detector.data_sources.first()
+    return ProjectUptimeSubscription.objects.get(uptime_subscription_id=int(data_source.source_id))
+
+
+def build_detector_fingerprint_component(detector: Any) -> str:
+    return f"uptime-detector:{detector.id}"
+
+
+def build_subscription_fingerprint_component(subscription_id: int) -> str:
+    return str(subscription_id)
+
+
+def hash_fingerprint(fingerprint: str) -> str:
+    return md5(fingerprint.encode("utf-8")).hexdigest()
+
+
+def update_auto_detected_active_interval_seconds(
+    apps: StateApps, schema_editor: BaseDatabaseSchemaEditor
+) -> None:
+    GroupHash = apps.get_model("sentry", "GroupHash")
+    DataSource = apps.get_model("workflow_engine", "DataSource")
+    ProjectUptimeSubscription = apps.get_model("uptime", "ProjectUptimeSubscription")
+
+    for data_source in RangeQuerySetWrapperWithProgressBar(
+        DataSource.objects.filter(type=DATA_SOURCE_UPTIME_SUBSCRIPTION)
+    ):
+        uptime_subscription_id = int(data_source.source_id)
+        project_subscription_id = ProjectUptimeSubscription.objects.get(
+            uptime_subscription_id=uptime_subscription_id
+        ).id
+        detector = data_source.detectors.first()
+
+        detector_fingerprint_hash = hash_fingerprint(build_detector_fingerprint_component(detector))
+        sub_fingerprint_hash = hash_fingerprint(
+            build_subscription_fingerprint_component(project_subscription_id)
+        )
+        group_hashes = list(
+            GroupHash.objects.filter(
+                project=detector.project,
+                hash__in=[detector_fingerprint_hash, sub_fingerprint_hash],
```

Comment:
Why even lookup the detector fingerprints? Don’t we only care about when it’s exactly just the sub hash?

Reply:
Oh I see, each group hash per group hash. Never mind